testing/sdk_metrics.py

```diff
@@ -22,17 +23,25 @@ def get_scheduler_metrics(service_name, timeout_seconds=15*60):
     """Returns a dict tree of Scheduler metrics fetched directly from the scheduler.
     Returned data will match the content of /service/<svc_name>/v1/metrics.
     """
+    @retrying.retry(
+        wait_fixed=1000,
+        stop_max_delay=timeout_seconds*1000,
+        retry_on_result=lambda res: not res)
```

Comment:
> This would retry for empty metrics too

Is that not the current case? See: https://github.com/dcos/shakedown/blob/master/shakedown/dcos/spinner.py#L40
``` 
if (not inverse_predicate and result) or (inverse_predicate and not result):
            count = count + 1
```
Given that inverse_predicate is False, if `not result` is True, the `result` is False and therefore the counter is not incremented and the call is retried.

Reply:
Using raise_for_status is almost always a good suggestion so I will definitely add that :)